spec/lib/rake/extensiontask_spec.rb

```diff
@@ -620,6 +620,102 @@
         spec.metadata['allowed_push_host'].should eq 'http://test'
       end
 
+      it "should set required_rubygems_version when building a gem for `-linux-gnu` or `-linux-musl`" do
+        if Gem::Version.new(Gem::VERSION) < Gem::Version.new("3.3.22")
+          skip "rubygems 3.3.22+ only"
+        end
+
+        platforms = ["arm64-darwin", "arm-linux", "arm-linux-gnu", "arm-linux-musl"]
+        ruby_cc_versions = ["3.3.0"]
+        ENV["RUBY_CC_VERSION"] = ruby_cc_versions.join(":")
+
+        ruby_cc_versions.each do |ruby_cc_version|
+          platforms.each do |platform|
+            rbconf = "/rubies/#{ruby_cc_version}/rbconfig.rb"
+            allow_any_instance_of(Rake::CompilerConfig).to(
+              receive(:find)
+                .with(ruby_cc_version, platform)
+                .and_return(rbconf)
+            )
+          end
+        end
+
+        allow(Gem).to receive_message_chain(:configuration, :verbose=).and_return(true)
+
+        spec = Gem::Specification.new do |s|
+          s.name = 'my_gem'
+          s.platform = Gem::Platform::RUBY
+          s.extensions = ['ext/somegem/extconf.rb']
+          s.metadata['allowed_push_host'] = 'http://test'
+        end
+
+        cross_specs = {} # platform => spec
+        Rake::ExtensionTask.new("extension_one", spec) do |ext|
+          ext.cross_platform = platforms
+          ext.cross_compile = true
+          ext.cross_compiling do |cross_spec|
+            cross_specs[cross_spec.platform.to_s] = cross_spec
+          end
+        end
+        platforms.each do |platform|
+          Rake::Task["native:my_gem:#{platform}"].execute
+        end
+
+        expected_rubygems_version = Gem::Requirement.new([">= 3.3.22"])
+
+        cross_specs["arm64-darwin"].required_rubygems_version.should eq(Gem::Requirement.default)
+        cross_specs["arm-linux"].required_rubygems_version.should eq(Gem::Requirement.default)
+        cross_specs["arm-linux-gnu"].required_rubygems_version.should eq(expected_rubygems_version)
+        cross_specs["arm-linux-musl"].required_rubygems_version.should eq(expected_rubygems_version)
+      end
+
+      it "should append to required_rubygems_version if it's set" do
+        if Gem::Version.new(Gem::VERSION) < Gem::Version.new("3.3.22")
+          skip "rubygems 3.3.22+ only"
+        end
+
```

Comment:
ditto.

Reply:
Similar to above: the platform `arm-linux-musl` _may_ be supported by rubygems (and so may end up as either `arm-linux` or `arm-linux-musl` once parsed), so the modified spec does not assert on the platform, and only asserts on the spec:

```ruby
        if Gem::Version.new(Gem::VERSION) < Gem::Version.new("3.3.22")
          actual_cross_spec.required_rubygems_version.should eq(Gem::Requirement.new(["!= 3.4.1"]))
        else
          actual_cross_spec.required_rubygems_version.should eq(expected_rubygems_version)
        end
```

I hope this is OK.

Reply:
I see.